### PR TITLE
Fixing invalid / misspelled / unknown allegiance codes in sector data

### PR DESCRIPTION
**Pull request by pmoule**
Sector data contains a lot of invalid / misspelled / unknown allegiance codes. In surveys this leads to a not complete _Alleg_ section in metadata. 

Expectation is, that used codes in a sector's world data should be displayed as allegiances in _Alleg_ section in sector data. As far as I can see, this is exactly your implementation.

Types of invalid codes are different. Sometimes these are simply misspelled wrt casing. Others are completely unknown and not defined in related sector xml. In a different case, allegiance code is different from the one in 'allegiance_code.tab' (e.g. _Nx_ and _NaXX_). In the latter case generally known code should be used.

**Examples:**
- [Typo: _IN_ should be _In_](https://travellermap.com/api/sec?sector=Markhab&type=SecondSurvey&milieu=M1105)
- [Typo: _Cc_ should be _Cr_](https://travellermap.com/api/sec?sector=Gvurrdon&type=SecondSurvey&milieu=M1248)
- [Missing unknown allegiances: _SI, DC, PA, TC_](https://travellermap.com/api/sec?sector=Ares&type=SecondSurvey&milieu=M1105)
- [Wrong code: _Nx_ instead of _NaXX_](https://travellermap.com/api/sec?sector=Bourne&type=SecondSurvey&milieu=M1105)

This pull request fixes [issue 240](https://github.com/inexorabletash/travellermap/issues/240)

### CODE REVIEW

File: res/Sectors/Distant Fringe/Abyss.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-0901 Masikaro             ?6A4???-? Fl Tz                               - - - ?22 10 Nx M3 V M7 V   
-1103 Gerund               ?512???-? Ic Tz                               - - - ?30 7  Nx M6 V        
-1706 Erim Arut            ?650???-? De He Po                            - - - ?30 11 Nx M7 III M2 V 
-2803 Boroyen              ?421???-? He Po Sa                            - - - ?01 6  Nx M8 V        
-2903 Darga                ?000???-? As Va                               - - - ?11 7  Nx M7 V M8 V    
-1211 Farisil              ?686???-? Ga Tz                               - - - ?10 7  Nx K1 V        
-1129 Allathur             ?6A3???-? Fl Tz                               - - - ?21 5  Nx M0 V        
-2336 Hamarah              ?432???-? Po Tz                               - - - ?10 8  Nx M2 V        
\ No newline at end of file
+0102 Jorutha              ?311???-? Ic Tz                               - - - ?11 8  NaXX M4 V
```
Comment (inexorabletash) [inline res/Sectors/Distant Fringe/Abyss.txt:3]: With the header untouched this is invalid. Make sure to use travellermap.com/tools/lintsec to validate any updated data.

File: res/Sectors/Faraway/Markhab.txt [hunk 2]
```diff
@@ -74,7 +74,7 @@ Picket                  1103  B242361-A  N Lo Mr Po    G 601 An K1 V M7 V
 Klathe                  1106  D879361-A    Lo Re       G 801 An K7 IV
 Vanzyke                 1110  D100261-A    Lo Re Va    G 301 An M8 VI DM
 Baza                    1111  C579363-B    Lo Re       G 223 An F6 V
-San Venaco              1122  E3433A7-5    Po Lo       G 230 IN K8 V M3 VI
+San Venaco              1122  E3433A7-5    Po Lo       G 230 In K8 V M3 VI
```
Comment (inexorabletash) [inline res/Sectors/Faraway/Markhab.txt:77]: Why this change?

File: res/Sectors/M1105/X'kug.txt [hunk 3]
```diff
@@ -41,11 +41,11 @@ Hex  Name                         UWP       Remarks
 0907 Ktuamat                      E777524-6 Ag Ni                                                            -       -  - 203   K4   K3 IV K5 V                          
 0909 Ltikrua                      B7B0669-B He Ni O:1107                                                     -       -  - 823   K4   M1 V M9 V                           
 0910 Naa Paghti                   C555A88-A Hi                                                               -       -  - 123   K4   G8 V                                
-1003 Tagn'akan                    A200557-D Ni Va GiruW                                                      -       K  - 404   Kc   M7 V M1 V                           
+1003 Tagn'akan                    A200557-D Ni Va GiruW                                                      -       K  - 404   Kk   M7 V M1 V
```
Comment (inexorabletash) [inline res/Sectors/M1105/X'kug.txt:44]: These are intentional K'kree client states.

**Review by inexorabletash — COMMENTED** [review]
This PR is too large to review. Please break it down into separate PRs for specific problems; ideally one per sector unless there are consistent issues across all files within a directory. 

Also, please ensure that all sector files have been validated using travellermap.com/tools/lintsec before submitting.

Also note that there's no requirement that 2-character allegiance codes used in legacy files be consistent across sectors. Be sure you're not making unnecessary changes. Include details about (1) the perceived problem (it may not be obvious, and/or it may not actually be a problem and (2) the proposed fix as part of the PR description.

**Comment by pmoule** [comment]
will break it down into smaller chunks for review

**Comment by pmoule** [comment]
closing for breaking down into smaller chunks
